### PR TITLE
Add speech client sample

### PR DESCRIPTION
**Pull request by jbolinger**
A simple speech example using the google-cloud-java client library in Kotlin.

/cc @neozwu 

Do not merge until all three samples have been reviewed and the top level `README.md` is updated.

### CODE REVIEW

File: speech/SpeechRecognitionClient/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+This directory contains an Android example that uses the
+[Cloud Speech-to-Text API](https://cloud.google.com/speech-to-text/)
+with the [Google Cloud Client Library for Java]
+
+## Prerequisites
+
+### Enable the Data Loss Prevention API
+
+If you have not already done so,
+[enable the Speech-to-Text API for your project](https://cloud.google.com/speech-to-text/docs/quickstart-client-libraries).
```
Comment (dzlier-gcp) [inline speech/SpeechRecognitionClient/README.md]: The link to enable this api in a project is https://pantheon.corp.google.com/flows/enableapi?apiid=speech.googleapis.com. I think this link should either change the wording (so it indicates to the user that it leads to the quickstart) or have the link go directly there.

Comment (jbolinger) [reply speech/SpeechRecognitionClient/README.md]: done (with https://console.cloud.google.com/flows/enableapi?apiid=speech.googleapis.com)

File: speech/SpeechRecognitionClient/app/src/androidTest/java/com/google/cloud/examples/speechrecognition/MainActivityTest.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Test
+    fun isReadyForSpeech() {
+        onView(allOf(withText(
+                getString(R.string.start_talking)), withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
+                .check(matches(isDisplayed()))
+    }
+
+    private fun getString(id: Int) =
+            InstrumentationRegistry.getTargetContext().resources.getString(id)
```
Comment (wisdal) [inline speech/SpeechRecognitionClient/app/src/androidTest/java/com/google/cloud/examples/speechrecognition/MainActivityTest.kt]: Is targetContext a property of InstrumentationRegistry?

File: speech/SpeechRecognitionClient/app/src/main/java/com/google/cloud/examples/speechrecognition/AudioEmitter.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+ */
+internal class AudioEmitter {
+
+    private var mAudioRecorder: AudioRecord? = null
+    private var mAudioExecutor: ScheduledExecutorService? = null
+    private lateinit var mBuffer: ByteArray
+
+    /** Start streaming  */
+    fun start(subscriber: (ByteString) -> Unit) {
+        // TODO: in a real app you may not want to fix these
```
[the remark this answers is not in the document]
Comment (wisdal) [reply speech/SpeechRecognitionClient/app/src/main/java/com/google/cloud/examples/speechrecognition/AudioEmitter.kt]: +1

File: speech/SpeechRecognitionClient/app/src/main/java/com/google/cloud/examples/speechrecognition/AudioEmitter.kt [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+        // start!
+        Log.d(TAG, "Recording audio with buffer size of: ${mBuffer.size} bytes")
+        mAudioRecorder!!.startRecording()
+
+        // stream bytes as they become available in chunks equal to the buffer size
+        mAudioExecutor!!.scheduleAtFixedRate({
+            // read audio data
+            val read = mAudioRecorder!!.read(
+                    mBuffer, 0, mBuffer.size, AudioRecord.READ_BLOCKING)
```
Comment (wisdal) [inline speech/SpeechRecognitionClient/app/src/main/java/com/google/cloud/examples/speechrecognition/AudioEmitter.kt]: Consider named parameters?

File: speech/SpeechRecognitionClient/README.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+resources as `app/src/main/res/raw/credential.json`.
+
+Again, ***you should not do this in your production app.***
+
+See the [Cloud Platform Auth Guide](https://cloud.google.com/docs/authentication#developer_workflow)
+for more information.
+
+## Client Libraries on Android
+
+The [Google Cloud Client Library for Java] that is used for this example can be used on Android,
```
[the remark this answers is not in the document]
Comment (jbolinger) [reply speech/SpeechRecognitionClient/README.md]: yup, updated.

File: speech/SpeechRecognitionClient/app/src/main/java/com/google/cloud/examples/speechrecognition/AudioEmitter.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (jbolinger) [reply speech/SpeechRecognitionClient/app/src/main/java/com/google/cloud/examples/speechrecognition/AudioEmitter.kt]: great idea, updated.

File: speech/SpeechRecognitionClient/app/src/androidTest/java/com/google/cloud/examples/speechrecognition/MainActivityTest.kt (hunk 2 above)
Comment (jbolinger) [reply speech/SpeechRecognitionClient/app/src/androidTest/java/com/google/cloud/examples/speechrecognition/MainActivityTest.kt]: nope, it's a static method.

File: speech/SpeechRecognitionClient/app/src/main/java/com/google/cloud/examples/speechrecognition/AudioEmitter.kt (hunk 4 above)
Comment (jbolinger) [reply speech/SpeechRecognitionClient/app/src/main/java/com/google/cloud/examples/speechrecognition/AudioEmitter.kt]: It's a Java function so I don't think that's possible?

Comment (wisdal) [reply speech/SpeechRecognitionClient/app/src/main/java/com/google/cloud/examples/speechrecognition/AudioEmitter.kt]: Yes, not possible if a Java function.

Comment (wisdal) [reply speech/SpeechRecognitionClient/app/src/main/java/com/google/cloud/examples/speechrecognition/AudioEmitter.kt]: Right! Not possible if a Java function.

File: speech/SpeechRecognitionClient/app/src/androidTest/java/com/google/cloud/examples/speechrecognition/MainActivityTest.kt (hunk 2 above)
Comment (jbolinger) [reply speech/SpeechRecognitionClient/app/src/androidTest/java/com/google/cloud/examples/speechrecognition/MainActivityTest.kt]: Yeah, that does look nicer but I believe that this is still a known limitation for tests defined in `androidTest`. Backticks can't be used there yet (my version of Android Studio flags it).

**Comment by frankyn** [comment]
@jbolinger the build is failing.

**Comment by frankyn** [comment]
Should we merge #71 first then?

**Comment by jbolinger** [comment]
@frankyn yes, it looks like it completed successfully and I think it'll fix the other two.

**Pull request merged**